aws-educate-tw/aws-educate-tpet-backend
# [SCRUM-163] Enhance `POST /send-email` API to support different recipient sources

Hi @chungchihhan, the work on [SCRUM-163](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-163?atlOrigin=eyJpIjoiNWI5N2FhMzdlZjc1NGY1NDlkN2Q5N2I4MTFjYjI4NDQiLCJwIjoiaiJ9) is complete.
Please review the code when you have a chance.

## Summary

This enhancement expands the `POST /send-email` API to support various recipient sources, streamlining email handling and ensuring compatibility with multiple recipient configurations.

## What's Changed
1. Added recipient_source field to specify how recipients should be provided to the backend.
  - `DIRECT`: Recipient emails and template variables are provided directly in the API request body.
  - `SPREADSHEET`: Recipient emails and template variables are provided via an uploaded spreadsheet file.
  
  
## Testing

### How to test
1. Make sure you are in the Postman `Shiun-local-dev` environment
2. Call the `POST /auth/login` API ([Postman link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/36968695-ac08ee18-f884-4f11-979b-f3dccaaa1bf0?action=share&source=copy-link&creator=36968695&ctx=documentation)) to get a valid JWT token in the **local-dev environment**
3. Call the `POST /send-email API` in the Postman `Shiun-local-dev` environment with a valid payload ([Postman API link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/35281217-5d2e366b-0768-424f-9653-66ae12ca6ba0?action=share&source=copy-link&creator=36968695&ctx=documentation))
4. You should receive a 202 Accepted response

### Test results

- [ img1 - Please see the Postman examples in `Email Service/Internal/Send Email` ([link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/35281217-5d2e366b-0768-424f-9653-66ae12ca6ba0?action=share&source=copy-link&creator=36968695&ctx=documentation)) ] 
  <img width="285" alt="image" src="https://github.com/user-attachments/assets/5af5e2bf-0d2c-44e7-b914-622c9a1c9eec">

- [ img2 - Email actually received, I have sent a copy of the same email to you @chungchihhan ]
  <img width="1460" alt="image" src="https://github.com/user-attachments/assets/9f8d3252-f698-431e-acf4-000793e74c1f">


## Additional Notes

1. **Make sure to call the APIs in the local-dev environment, and use Postman APIs under the Internal folder.**

related issue: [SCRUM-163](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-163?atlOrigin=eyJpIjoiNWI5N2FhMzdlZjc1NGY1NDlkN2Q5N2I4MTFjYjI4NDQiLCJwIjoiaiJ9)

[SCRUM-163]: https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
